pkg/network/state.go

```diff
@@ -326,31 +335,46 @@ func (ns *networkState) StoreClosedConnection(conn *ConnectionStats) {
 }
 
 // storeDNSStats stores latest DNS stats for all clients
-func (ns *networkState) storeDNSStats(stats map[DNSKey]map[string]DNSStats) {
-	for key, statsByDomain := range stats {
-		for _, client := range ns.clients {
-			// If we've seen DNS stats for this key already, let's combine the two
-			if prevByDomain, ok := client.dnsStats[key]; ok {
-				for domain, dns := range statsByDomain {
-					if prev, ok := prevByDomain[domain]; ok {
+func (ns *networkState) storeDNSStats(stats map[DNSKey]map[string]map[layers.DNSType]DNSStats) {
+	for _, client := range ns.clients {
+		dnsStatsThisClient := 0
```

Comment:
This counter is assuming that `client.dnsStats` starts out empty. Is that guaranteed at this point?